// @dgcode/sites-app/dist/classes/Comment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Exports -----------------------------------
/**
 * A Comment attached to any Sites page.
 */
class Comment {
    /**
     * Deletes this comment.
     */
    deleteComment() {
    }
    /**
     * Gets the email address of the author of this comment.
     *
     * @returns {String} the author's email
     */
    getAuthorEmail() {
        const finalString = null;
        return finalString;
    }
    /**
     * Gets the name of the author of this comment.
     *
     * @returns {String} the author's name
     */
    getAuthorName() {
        const finalString = null;
        return finalString;
    }
    /**
     * Return the content of this comment as a String.
     *
     * @returns {String} the comment content
     */
    getContent() {
        const finalString = null;
        return finalString;
    }
    /**
     * Return the date this comment was originally published.
     *
     * @returns {Date} the date of original publication
     */
    getDatePublished() {
        const date = null;
        return date;
    }
    /**
     * Return the date this comment was last updated.
     *
     * @returns {Date} the last updated date
     */
    getLastUpdated() {
        const date = null;
        return date;
    }
    /**
     * Get the parent page of this comment.
     *
     * @returns {Page} the parent page
     * @see https://developers.google.com/apps-script/reference/sites/page.html
     */
    getParent() {
        const page = null;
        return page;
    }
    /**
     * Set the content of this comment.
     *
     * @param {string} content the new content
     * @returns {Comment} this Comment for chaining
     */
    setContent(content) {
        return this;
    }
    /**
     * Set the parent page of this comment.
     *
     * @param {Page} parent the new parent
     * @returns {Comment} this Comment for chaining
     */
    setParent(parent) {
        return this;
    }
}
exports.Comment = Comment;
