// @dgcode/sites-app/dist/classes/ListItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Exports -----------------------------------
/**
 * A Sites ListItem - a list element from a Sites List page.
 */
class ListItem {
    /**
     * Deletes this list item.
     */
    deleteListItem() {
    }
    /**
     * Return the date this list item was first published.
     *
     * @returns {Date} the date of original publication
     */
    getDatePublished() {
        const date = null;
        return date;
    }
    /**
     * Return the date this comment was last updated.
     *
     * @returns {Date} the last updated date
     */
    getLastUpdated() {
        const date = null;
        return date;
    }
    /**
     * Get the parent page of this list item.
     *
     * @returns {Page} the parent page
     * @see https://developers.google.com/apps-script/reference/sites/page.html
     */
    getParent() {
        const page = null;
        return page;
    }
    /**
     * Get the value of this ListItem for a numbered column.
     *
     * @param {number} index the column to get the value of
     * @returns {String} the value of that column
     */
    getValueByIndex(index) {
        const finalString = null;
        return finalString;
    }
    /**
     * Get the value of this ListItem for a named column.
     *
     * @param {string} name the column to get the value of
     * @returns {String} the value of that column
     */
    getValueByName(name) {
        const finalString = null;
        return finalString;
    }
    /**
     * Set the parent page of this list item.
     *
     * @param {Page} parent the new parent
     * @returns {ListItem} this ListItem for chaining
     */
    setParent(parent) {
        return this;
    }
    /**
     * Set the value of this ListItem for a numbered column.
     *
     * @param {number} index the column to set the value of
     * @param {string} value the new value
     * @returns {ListItem} this ListItem for chaining
     */
    setValueByIndex(index, value) {
        return this;
    }
    /**
     * Set the value of this ListItem for a numbered column.
     *
     * @param {string} name the column to set the value of
     * @param {string} value the new value
     * @returns {ListItem} this ListItem for chaining
     */
    setValueByName(name, value) {
        return this;
    }
}
exports.ListItem = ListItem;
